Sources/Basics/HTTPClient.swift

```diff
@@ -87,6 +87,9 @@ public struct HTTPClient: Cancellable {
         progress: ProgressHandler? = nil,
         completion: @escaping CompletionHandler
     ) {
+        guard configuration.isEnabled else {
+            return completion(.failure(CancellationError()))
```

Comment:
AFAICT this is the place where errors are thrown when the network is disabled — probably a more specific error that describes the reason for the cancellation would be useful.  I realize this PR is still in Draft so maybe that was already planned, but just a thought for when this becomes ready for review.